### PR TITLE
FIX fetch all list multicompany no transverse mode

### PR DESCRIPTION
**Pull request by ptibogxiv**

### CODE REVIEW

File: htdocs/user/class/user.class.php [hunk 1]
```diff
@@ -3214,7 +3214,7 @@ function fetchAll($sortorder='', $sortfield='', $limit=0, $offset=0, $filter=arr
 
 		$sql="SELECT t.rowid";
 		$sql.= ' FROM '.MAIN_DB_PREFIX .$this->table_element.' as t ';
-		$sql.= " WHERE 1";
+		$sql.= " WHERE t.entity IN (".getEntity('user').")";
```
Comment (hregis) [inline htdocs/user/class/user.class.php:3217]: @ptibogxiv 
maybe send this in "$filter" parameter when you need it?

Comment (ptibogxiv) [reply htdocs/user/class/user.class.php:3217]: @hregis i want  in no transverse mode, only display users of the entity in email form

Comment (hregis) [reply htdocs/user/class/user.class.php:3217]: @ptibogxiv 
this breaks the function "fetchAll" as the name suggests! In addition it is not compatible with the "transversal" mode.

Comment (hregis) [reply htdocs/user/class/user.class.php:3217]: @ptibogxiv @eldy 
can you try with this patch? : https://github.com/Dolibarr/dolibarr/pull/11060

Comment (ptibogxiv) [reply htdocs/user/class/user.class.php:3217]: #11060 is good for me, better select of user by entity

**Pull request closed without merging**